packages/espressocash_app/lib/features/kyc_sharing/services/kyc_service.dart

```diff
@@ -0,0 +1,206 @@
+  }
+
+  Future<void> updateBasicInfo({
+    String? firstName,
+    String? lastName,
+    DateTime? dob,
+    String? idNumber,
+    DocumentType? idType,
+    String? countryCode,
+  }) async {
+    await _kycRepository.updateUserData(
+      name: Name(
+        firstName: firstName ?? '',
+        lastName: lastName ?? '',
+        id: value?.name?.first.id ?? '',
+      ),
+      birthDate: dob?.let(
+        (e) => BirthDate(
+          value: e,
+          id: value?.birthDate?.first.id ?? '',
+        ),
+      ),
+      document: Document(
+        type: idType?.toIdType() ?? IdType.other,
+        number: idNumber ?? '',
+        countryCode: countryCode ?? '',
+        id: value?.document?.first.id ?? '',
+      ),
+    );
+
+    // TODO(dev): move selecting of partner, should not be here
+    await _kycRepository.grantPartnerAccess(partnerAuthPk);
+
+    await fetchUserData();
+  }
+
+  Future<void> updateBankInfo({
+    required String bankAccountNumber,
+    required String bankCode,
+    String? bankName,
+  }) async {
+    await _kycRepository.updateUserData(
+      bankInfo: BankInfo(
+        accountNumber: bankAccountNumber,
+        bankCode: bankCode,
+        bankName: bankName ?? '',
+        id: value?.bankInfo?.first.id ?? '',
+      ),
+    );
+
+    await fetchUserData();
+  }
+
+  Future<void> initDocumentValidation() async {
+    await _kycRepository.initKycVerification(
+      nameId: value?.name?.first.id ?? '',
+      birthDateId: value?.birthDate?.first.id ?? '',
+      documentId: value?.document?.first.id ?? '',
+      selfieImageId: value?.selfie?.first.id ?? '',
+    );
+
+    await fetchUserData();
+    _subscribe();
+  }
+
+  Future<void> updateSelfiePhoto({File? photoSelfie}) async {
+    await _kycRepository.updateUserData(
+      selfie: photoSelfie != null
+          ? Selfie(
+              value: await photoSelfie.readAsBytes(),
+              id: value?.selfie?.first.id ?? '',
+            )
+          : null,
+    );
+
+    await fetchUserData();
+  }
+
+  Future<void> initEmailVerification({required String email}) async {
+    await _kycRepository.grantPartnerAccess(validatorAuthPk);
+
+    await _kycRepository.updateUserData(
+      email: Email(
+        value: email,
+        id: value?.email?.first.id ?? '',
+      ),
+    );
+
+    await fetchUserData();
+
+    await _kycRepository.initEmailVerification(
+      emailId: value?.email?.first.id ?? '',
+    );
+  }
+
+  Future<void> verifyEmail({required String code}) async {
+    await _kycRepository.verifyEmail(
+      code: code,
+      dataId: value?.email?.first.id ?? '',
+    );
+
+    await fetchUserData();
+  }
+
+  Future<void> initPhoneVerification({required String phone}) async {
+    await _kycRepository.updateUserData(
+      phone: Phone(
+        value: phone,
+        id: value?.phone?.first.id ?? '',
+      ),
+    );
+
+    await fetchUserData();
+
+    // TODO(vsumin): Currently do not work on test environment
```

Comment:
It should be working now, just doesn't send the code.

Reply:
Removed